src/Raven.Server/Documents/Queries/AST/Query.cs

```diff
@@ -327,123 +328,159 @@ public enum Type
         }
 
         public Type Aggregation;
+        public bool Any => _numberOfValues > 0;
 
-        private double _val;
-        private int _valIndex;
-        public long Count;
+        private object[] _values;
+        private object[] _count;
 
-        public TimeSeriesAggregation(int valIndex, TimeSeriesAggregation.Type type)
+        public object[] Count
+        {
+            get
+            {
+                Array.Resize(ref _count, _numberOfValues);
+                return _count;
+            }
+        }
+
+        private byte _numberOfValues;
+
+        public TimeSeriesAggregation(TimeSeriesAggregation.Type type)
         {
-            _valIndex = valIndex;
             Aggregation = type;
-            _val = 0;
-            Count = 0;
+            _count = new object[TimeSeriesValuesSegment.MaxNumberOfValues];
```

Comment:
I'm wondering if we should use some kind of an array pool for this. From the other side this is cached in query cache?

Reply:
This is fairly constant, I don't think we are going to need more. 
Especially because it will be complex to return them.
That said, why not use a List here?

Reply:
no good reason.
changed to List